test/e2e/backup_s3compatible_test.go

```diff
@@ -0,0 +1,317 @@
+
+	// Get S3 secrets and provide it to minio server
+	secret, err := f.KubeClient.Core().Secrets(f.Namespace).Get(os.Getenv("TEST_AWS_SECRET"), metav1.GetOptions{})
+
+	if err != nil {
+		t.Fatal("an error occurred while reading aws secret")
+	}
+
+	credsini := secret.Data[api.AWSSecretCredentialsFileName]
+
+	// Read AWS configuration file
+	cfg, err := ini.Load(credsini)
+
+	// Extract keys
+	accessKey := cfg.Section("default").Key("aws_access_key_id").String()
+	secretAccessKey := cfg.Section("default").Key("aws_secret_access_key").String()
+
+	// Minio pod
+	minioPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "minio",
+			Labels: map[string]string{
+				"app": "minio",
+			},
+		},
+		Spec: v1.PodSpec{
+			Volumes: []v1.Volume{
+				{
+					Name: "storage",
+					VolumeSource: v1.VolumeSource{
+						EmptyDir: &v1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+			InitContainers: []v1.Container{
+				{
+					Name:  "create-bucket",
+					Image: "busybox",
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "storage",
+							MountPath: "/storage",
+						},
+					},
+					Command: []string{
+						"mkdir",
+						"-p",
+						fmt.Sprintf("%s/%s", "/storage", os.Getenv("TEST_S3_BUCKET")),
+					},
+				},
+			},
+			Containers: []v1.Container{
+				{
+					Name:            "minio",
+					Image:           "minio/minio",
+					ImagePullPolicy: v1.PullIfNotPresent,
+					Args: []string{
+						"server",
+						"/storage",
+					},
+					Env: []v1.EnvVar{
+						{
+							Name:  "MINIO_ACCESS_KEY",
+							Value: accessKey,
+						},
+						{
+							Name:  "MINIO_SECRET_KEY",
+							Value: secretAccessKey,
+						},
+					},
+					Ports: []v1.ContainerPort{
+						{
+							ContainerPort: 9000,
+						},
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "storage",
+							MountPath: "/storage",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Minio service
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "minio",
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{
+				"app": "minio",
+			},
+			Ports: []v1.ServicePort{
+				{
+					Port:       9000,
+					TargetPort: intstr.FromInt(9000),
+				},
+			},
+		},
+	}
+
+	f.KubeClient.Core().Services(f.Namespace).Create(svc)
+
+	_, err = f.KubeClient.Core().Pods(f.Namespace).Create(minioPod)
+	if err != nil {
+		t.Fatal("an error occurred while running minio pod")
+	}
+
+	// Wait for minio pod
+	interval := 5 * time.Second
+	err = retryutil.Retry(interval, int(30*time.Second/(interval)), func() (bool, error) {
+		pod, err := f.KubeClient.Core().Pods(f.Namespace).Get(minioPod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if pod.Status.Phase == v1.PodRunning {
+			return true, nil
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		t.Fatalf("failed to wait minio pods running: %v", err)
+	}
+
+}
+
+// teardownMinio deletes minio server pod
+func teardownMinio(t *testing.T) {
```

Comment:
I've forgotten minio service in this part.

Reply:
?

Reply:
it only deletes the pod, not the services i guess.

Reply:
exactly!